providers/flipt/pom.xml

```diff
@@ -36,12 +36,32 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-core</artifactId>
+            <version>2.16.1</version>
+            <scope>test</scope>
```

Comment:
@markphelps removing the test scope from all 3 jackson dependencies clears the warning and makes the CI pass.
This adds runtime dependencies and not completely fit [Flipt Java client dependencies](https://github.com/flipt-io/flipt-server-sdks/blob/main/flipt-java/build.gradle#L29), can be ok if ok by you too.

@toddbaert

Reply:
@markphelps I think @liran2000 may be on to something. I _think_ you may actually need [some jackson deps](https://github.com/search?q=repo%3Aflipt-io%2Fflipt-server-sdks%20com.fasterxml.jackson.annotation&type=code) as runtime dependencies. I've confirmed what he said above. When the javadoc gen runs at least, those classes aren't available...

If you're sure this isn't a real issue, we can merge [this](https://github.com/open-feature/java-sdk-contrib/pull/645).

Reply:
Thanks @toddbaert and @liran2000 ! I think you are correct that we are probably missing `jackson-annotations` being declared as a dependency in the new `flipt-java` lib. 

My Java days are (mostly) behind me, so I just want to verify that adding `com.fasterxml.jackson.core:jackson-annotations:2.16.1` to our [build.gradle](https://github.com/flipt-io/flipt-java/blob/main/build.gradle#L16C1-L18C72) and creating a new release should fix the issue here?

I guess I could build a snapshot or local version of the `flipt-java` lib with this change, then update it here to test that the warnings go away

Reply:
I think you can try adding runtime dependencies for [jackson-databind](https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind/2.16.1) on flipt java instead of all your Jackson dependencies, it contains compile dependencies for [jackson-core](https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-core) and [jackson-annotations](https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-annotations):
```
implementation 'com.fasterxml.jackson.core:jackson-databind:2.16.1'
```

if it doesn't work, [jackson-core](https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-core) and [jackson-annotations](https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-annotations) can be added as well, but I don't think it's needed.

Reply:
Not sure why Javadoc generation is still failing, it is passing locally for me. The allegadly missing classes are included in the provider project. Maybe adding all 3 Jackson dependencies can solve it ?
@toddbaert thoughts on why it is failing ? or merge your [PR](https://github.com/open-feature/java-sdk-contrib/pull/645) ?
![Uploading image.png…]()

Reply:
I was experimenting and still to see if perhaps this was a caching issue. I think it's not. Perhaps it's a JVM version thing? I'm certainly OK with merging [this](https://github.com/open-feature/java-sdk-contrib/pull/645) if @markphelps is.

Reply:
sorry i mean to look at this. just hit approve

Reply:
> sorry i mean to look at this. just hit approve

Cool, you should be able to merge/rebase-on main and this should pass :crossed_fingers:

Reply:
@toddbaert done!